### PR TITLE
java17 upgrade

### PR DESCRIPTION
**Pull request by SandhyaBhatia**
# Summary of Changes

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Public API Additions/Changes

Please include a description of any new or modified publicly available classes and/or methods (if applicable).

## Downstream Consumer Impact

Please include a description of how downstream consumers will be affected (breaking changes, bug fixes, internal behavior modification, etc.). If this PR introduces breaking changes please list out each of the breaking changes, as well as steps for migrating incompatible code.

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Test A
- [X] Test B

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by rodohanna — DISMISSED** [review]
This is demo.

**Comment by rodohanna** [comment]
Would you mind updating all of the PR description sections?

File: build.gradle [hunk 1]
```diff
@@ -8,19 +8,23 @@ plugins {
 }
 
 group "com.mx.coppuccino"
-version "2.0.0"
+version "2.0.1"
```
Comment (rodohanna) [inline build.gradle:11]: Let's the bump the version in a new PR to keep the version history visible in the commit history.

File: build.gradle [hunk 2]
```diff
@@ -8,19 +8,23 @@ plugins {
...
 }
 
 dependencies {
-    implementation "ru.vyarus:gradle-quality-plugin:4.3.0"
-    implementation "com.github.spotbugs:spotbugs-gradle-plugin:3.0.0"
-    implementation "com.diffplug.spotless:spotless-plugin-gradle:6.5.2"
+    implementation "ru.vyarus:gradle-quality-plugin:4.7.0"
+    implementation 'com.github.spotbugs.snom:spotbugs-gradle-plugin:5.0.6'
+    implementation ("com.diffplug.spotless:spotless-plugin-gradle:6.6.1") {
+        exclude group: 'com.google.googlejavaformat', module: 'google-java-format-parent'
```
Comment (rodohanna) [inline build.gradle:25]: can we use double quotes on this line too? to keep everything consistent.

File: build.gradle [hunk 3]
```diff
@@ -8,19 +8,23 @@ plugins {
 }
 
 group "com.mx.coppuccino"
-version "2.0.0"
+version "2.0.1"
 
 sourceCompatibility = 1.8
+targetCompatibility = 1.8
```
Comment (rodohanna) [inline build.gradle:14]: thanks for adding this!

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -3,4 +3,4 @@ distributionPath=wrapper/dists
 distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.1-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-
+org.gradle.java.home= /Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home
```
Comment (rodohanna) [inline gradle/wrapper/gradle-wrapper.properties:6]: this should probably be removed.

**Comment by SandhyaBhatia** [comment]
> Would you mind updating all of the PR description sections?

Yes!

File: gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (SandhyaBhatia) [reply gradle/wrapper/gradle-wrapper.properties:6]: > this should probably be removed.

on it

**Review by toledobh — APPROVED** [review]
Looks good, let's upgrade 🌟

**Comment by SandhyaBhatia** [comment]
> This is demo.

done

**Review by RyanJones08 — APPROVED** [review]
Looks good

**Pull request merged**